evm/p2p/kademlia.py

```diff
@@ -264,6 +268,20 @@ def neighbours(self, node_id, k=k_bucket_size):
         return sort_by_distance(nodes, node_id)[:k]
 
 
+def binary_get_bucket_for_node(buckets, node):
+    """Return the bucket for a given node."""
+    sorted_buckets = sorted(buckets)
+    bucket_ends = list(KBucket.end for KBucket in sorted_buckets)
+    bucket_position = bisect.bisect_left(bucket_ends, node.id)
+    # Prevents edge cases where bisect_left returns an out of range index
+    if bucket_position >= len(buckets):
```

Comment:
It's functionally the *same* but I like it better because it optimizes for the success case rather than adding overhead pre-checking for the failure case.

```python
try:
    bucket = sorted_buckets[bucket_position]
except IndexError:
    raise ValueError("No bucket found for node with id {}".format(node.id))
```

Reply:
good to know, i thought about doing this but couldn't think of a good reason to, now i do!